src/configure/helper/telemetryHelper.ts

```diff
@@ -0,0 +1,76 @@
+import { IActionContext, ITelemetryReporter, parseError } from "vscode-azureextensionui";
+
+import { extensionVariables } from "../model/models";
+import { TelemetryKeys } from '../resources/telemetryKeys';
+import * as logger from '../../logger';
+
+const uuid = require('uuid/v4');
+
+export class TelemetryHelper {
+    private actionContext: IActionContext;
+    private telemetryReporter: ITelemetryReporter;
+    private journeyId: string;
+    private command: string;
+    constructor(actionContext: IActionContext, command: string) {
+        this.actionContext = actionContext;
+        this.telemetryReporter = extensionVariables.reporter;
+        this.journeyId = uuid();
+        this.command = command;
+        this.setTelemetry(TelemetryKeys.Command, command);
+        this.setTelemetry(TelemetryKeys.JourneyId, this.journeyId);
+    }
+
+    public setTelemetry(key: string, value: string) {
+        if (key) {
+            this.actionContext.telemetry.properties[key] = value;
+        }
+    }
+
+    public setResult(result: 'Succeeded' | 'Failed' | 'Canceled', error?: Error) {
+        this.actionContext.telemetry.properties.result = result;
+        if (error) {
+            let parsedError = parseError(error);
+            this.actionContext.telemetry.properties.error = JSON.stringify(parsedError);
+            this.actionContext.telemetry.properties.errorMessage = parsedError.message;
+        }
+    }
+
+    public setCurrentStep(stepName: string) {
+        this.actionContext.telemetry.properties.cancelStep = stepName;
```

Comment:
Minor: do you want to call this property currentStep ? This way if there is error or operation is cancelled, we know what the current step is.

Reply:
This property already exists in the telemetry.properties, hence reused it.